### PR TITLE
test: add unit test to plugin

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR adds unit test to plugin

### CODE REVIEW

**Comment by Ian2012** [comment]
@BryanttV Please omit settings files and add tests for forum_digest command

**Review by Ian2012 — CHANGES_REQUESTED** [review]
@BryanttV Please omit settings files and add tests for the `forum_digest` command

File: platform_plugin_forum_email_notifier/tests/test_filters.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+""" Unit tests for filters in `platform_plugin_forum_email_notifier` plugin."""
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from platform_plugin_forum_email_notifier.extensions.filters import AddInstructorNotifierTab
+from platform_plugin_forum_email_notifier.models import PreferenceOptions
+
+module_path = "platform_plugin_forum_email_notifier.extensions.filters"
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_filters.py]:
```suggestion
FILTERS_MODULE_PATH = "platform_plugin_forum_email_notifier.extensions.filters"
```

File: platform_plugin_forum_email_notifier/tests/test_filters.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        mock_get.return_value.preference = PreferenceOptions.ALL_POSTS
+        context = {"course": Mock(id="test-course-id"), "sections": []}
+        template_name = "test-template-name"
+
+        result = self.filter.run_filter(context, template_name)
+
+        self.assertEqual(
+            result["context"]["current_preference"], PreferenceOptions.ALL_POSTS
+        )
+        self.assertEqual(len(result["context"]["sections"]), 1)
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_filters.py:48]: how do we know this section is the instructor tab one?

File: platform_plugin_forum_email_notifier/tests/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        mock_filter.return_value.exists.return_value = False
+        context = {"course": Mock(id="test-course-id"), "sections": []}
+        template_name = "test-template-name"
+
+        result = self.filter.run_filter(context, template_name)
+
+        self.assertEqual(
+            result["context"]["current_preference"], PreferenceOptions.NONE
+        )
+        self.assertEqual(len(result["context"]["sections"]), 1)
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_filters.py:76]: Same here

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+from platform_plugin_forum_email_notifier.tasks import (
+    handle_digests,
+    notify_users,
+    send_digest,
+    send_email_notification,
+)
+from platform_plugin_forum_email_notifier.utils import ForumObject
+
+User = get_user_model()
+module_path = "platform_plugin_forum_email_notifier.tasks"
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]: TASKS_MODULE_PATH

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+        """
+        mock_user = Mock(spec=User, id=1, email="test@user-email.com")
+        mock_get.return_value = mock_user
+        mock_get_course_overview_or_none.return_value = Mock()
+        mock_get_user_preference.return_value = "en"
+
+        send_email_notification(**self.send_email_notification_args)
+
+        mock_send_forum_email_notification.assert_called_once_with(
+            recipient=Recipient(1, "test@user-email.com"),
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]:
```suggestion
            recipient=Recipient(mock_user.id, mock_user.email),
```

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+    @send_forum_email_notification_mock
+    def test_send_email_notification_user_exists(
+        self,
+        mock_send_forum_email_notification: Mock,
+        mock_get_user_preference: Mock,
+        mock_get_course_overview_or_none: Mock,
+        mock_get: Mock,
+    ):
+        """
+        Check `send_email_notification` when user exists.
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]: Can we elaborate a bit more?

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 7]
```diff
@@ -0,0 +1,527 @@
...
+                "author_username": "test-author-username",
+                "author_email": "test@author-email.com",
+                "object_type": ForumObject.THREAD,
+            },
+        )
+
+    @get_mock
+    def test_send_email_notification_user_does_not_exist(self, mock_get: Mock):
+        """
+        Check `send_email_notification` when user does not exist.
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]: If we're gonna use `Check...` let's add `Check send_email_notification behavior...`

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+                "author_email": "test@author-email.com",
+                "object_type": ForumObject.THREAD,
+            },
+        )
+
+    @get_mock
+    @get_course_overview_or_none_mock
+    @get_user_preference_mock
+    @send_forum_email_notification_mock
+    def test_send_email_notification_post_does_not_have_title(
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]: can we use ddt?

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 9]
```diff
@@ -0,0 +1,527 @@
...
+            url="test-url/",
+            author_id="test-author_id",
+            author_username="test-author_username",
+            author_email="test@author_email.com",
+            object_type=ForumObject.THREAD,
+            context={},
+        )
+
+        mock_send_email_notification.assert_called()
+        mock_handle_digests.assert_called()
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]: can we check the arguments are correct?

File: platform_plugin_forum_email_notifier/tests/test_tasks.py [hunk 10]
```diff
@@ -0,0 +1,527 @@
...
+        handle_digests(**self.handle_digest_args)
+
+        mock_filter.assert_called_once_with(
+            preference__in=(
+                PreferenceOptions.ALL_POSTS_DAILY_DIGEST,
+                PreferenceOptions.ALL_POSTS_WEEKLY_DIGEST,
+            ),
+            course_id=self.handle_digest_args["course_id"],
+        )
+
```
Comment (mariajgrimaldi) [inline platform_plugin_forum_email_notifier/tests/test_tasks.py]:
```suggestion
```

**Review by mariajgrimaldi — APPROVED** [review]
LGTM. Just a few comments left to address

**Review by Ian2012 — APPROVED** [review]

**Pull request merged**